lib/vtls/openssl.c

```diff
@@ -2031,6 +2031,10 @@ static CURLcode ossl_set_provider(struct Curl_easy *data, const char *iname)
     data->state.libctx = libctx;
   }
 
+  if(!OSSL_LIB_CTX_load_config(data->state.libctx, NULL)) {
+    infof(data, "Failed to load default openssl config. Proceeding.");
+  }
```

Comment:
This seems to opens up curl to the vulnerability in OpenSSL where
it opens world-writable config files in certain situations (e.g. in its
default builds), on Windows. Then such world-writable config file can
inject an attacker DLL to the process.

I haven't dug into the details of this particular case, if it's
somehow exempted from the issue.

Aside from fixing OpenSSL (the chance for this seems almost zero),
could this code respect the
`CURL_DISABLE_OPENSSL_AUTO_LOAD_CONFIG` feature guard 
to allow creating a non-vulnerable build?